lib/monetary.ts

```diff
@@ -34,50 +34,35 @@ export class Ethereum implements Currency {
   }
 
   get name(): string {
-    return 'ethereum';
+    return 'Ethereum';
   }
 }
 
 export class ERC20 implements Currency {
-  constructor(protected _name: string, protected _decimals: number = 18) {}
-
-  get decimals(): number {
-    return this._decimals;
-  }
-
-  get name(): string {
-    return this._name;
-  }
+  constructor(
+    readonly name: string,
+    readonly address: string,
+    readonly decimals: number = 18
+  ) {}
 }
 
 export class Tether extends ERC20 implements Currency {
```

Comment:
We might want to integrate with the Uniswap token list in the future: https://github.com/Uniswap/token-lists